packages/taquito-core/src/errors.ts

```diff
@@ -264,3 +264,15 @@ export class ProhibitedActionError extends UnsupportedActionError {
     this.name = 'ProhibitedActionError';
   }
 }
+
+/**
+ *  @category Error
+ *  @description Error indicates a failure in grabbing the public key
+ */
+export class PublicKeyNotFoundError extends TaquitoError {
+  constructor(public pkh: string, public cause?: any) {
```

Comment:
I also suggest that we make public fields `readonly`

Reply:
unsure how much impact it will be but doesn't hurt to add so updated all error classes accordingly